pkg/diagnostics/workflow_metrics.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package diagnostics
+
+import (
+	"context"
+
+	diagUtils "github.com/dapr/dapr/pkg/diagnostics/utils"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	elapsedKey       = tag.MustNewKey("elapsed")
+	reminderTypeKey  = tag.MustNewKey("reminder_type")
+	executionTypeKey = tag.MustNewKey("execution_type")
+)
+
+const (
+	StatusSuccess   = "success"
+	StatusFailed    = "failed"
+	StatusRetryable = "retryable"
+	CreateWorkflow  = "create_workflow"
+	GetWorkflow     = "get_workflow"
+	AddEvent        = "add_event"
+	PurgeWorkflow   = "purge_workflow"
+	Activity        = "activity"
+	Workflow        = "workflow"
+
+	WorkflowEvent = "event"
+	Timer         = "timer"
+
+	ComponentName = "dapr"
+)
+
+type workflowMetrics struct {
+	// Total Successful and Failed workflow operation requests
+	workflowOperationsCount *stats.Int64Measure
```

Comment:
So, it's total number of requests.
Is there any metric for `failed_total` as well, just like mentioned in https://github.com/dapr/dapr/issues/7109#issue-1967760866

Reply:
It is a collective count of requests created, if a request fails due to some reasons, a key-value pair of `status: failed` is attached to record. like [here](https://github.com/dapr/dapr/pull/7152/files#diff-9d8a4ceb7b0a3e0bb6d0ec2d8527ea7771d223ba416c55e727fc950b52b68f10R200)